mindsdb/interfaces/agents/langchain_agent.py

```diff
@@ -367,41 +388,70 @@ def run_agent(self, df: pd.DataFrame, agent: AgentExecutor, args: Dict) -> pd.Da
 
         def _invoke_agent_executor_with_prompt(agent_executor, prompt):
             if not prompt:
-                return ''
+                return {'context': [], 'answer': ''} if return_context else ''
             try:
-                # Handle callbacks per run.
-                answer = agent_executor.invoke(prompt, config={'callbacks': self._get_agent_callbacks(args)})
+                context_callback = ContextCaptureCallback()
+                callbacks = self._get_agent_callbacks(args)
+                callbacks.append(context_callback)
+
+                result = agent_executor.invoke(prompt, config={'callbacks': callbacks})
+
+                if return_context:
+                    captured_context = context_callback.get_contexts()
+                    if isinstance(result, dict) and 'output' in result:
+                        output = result['output']
+                        return {'context': captured_context, 'answer': output}
+                    else:
+                        return {'context': captured_context, 'answer': str(result)}
+                else:
+                    return result['output'] if isinstance(result, dict) and 'output' in result else str(result)
```

Comment:
returned value can be string or dict, I think it can lead to some issues with expecting different type.